naga/src/back/msl/writer.rs

```diff
@@ -2768,7 +2768,9 @@ impl<W: Write> Writer<W> {
                     if !continuing.is_empty() || break_if.is_some() {
                         let gate_name = self.namer.call("loop_init");
                         writeln!(self.out, "{level}bool {gate_name} = true;")?;
-                        writeln!(self.out, "{level}while(true) {{")?;
+                        // The empty inline assembly "call" will prevent the metal compiler
+                        // from optimizing out these potentially-infinte loops.
+                        writeln!(self.out, "{level}while(true) {{__asm__(\"\");")?;
```

Comment:
idea(non-blocking): I wonder if this might be easier for a Rustacean to understand if this had a helper a la [`std::hint::black_box`](https://doc.rust-lang.org/stable/std/hint/fn.black_box.html)?

Reply:
And the helper would inject `__asm__("");`? That would work but I think that for now it's clearer that only loops use this hack.